src/vm/node_modules/VM.js

```diff
@@ -8799,10 +8799,23 @@ function installZone(payload, log, callback)
         }, function (cb) {
             // var zoneinit is in installZone() scope
 
-            // when receiving zoneinit is never run.
+            // When receiving zoneinit is never run, instead detach/attach the
+            // zone to ensure the cores dataset gets created properly by the
+            // brand scripts.
             if (receiving) {
-                cb();
-                return;
+                vasync.pipeline(
+                    {funcs: [
+                        function (_, cb2) {
```

Comment:
These are in a vasync.pipeline so they will run sequentially, right?

Reply:
Correct.